sql-plugin/src/main/delta-lake-common/scala/com/nvidia/spark/rapids/delta/RapidsDeltaWrite.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.delta
+
+import com.nvidia.spark.rapids.{BaseExprMeta, DataFromReplacementRule, DataWritingCommandMeta, GpuExec, GpuMetric, GpuOverrides, PartMeta, RapidsConf, RapidsMeta, ScanMeta, SparkPlanMeta}
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, UnaryNode}
+import org.apache.spark.sql.execution.{SparkPlan, SparkStrategy, UnaryExecNode}
+import org.apache.spark.sql.execution.datasources.v2.V2CommandExec
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.rapids.GpuWriteJobStatsTracker
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+/** Strategy rule to translate a logical RapidsDeltaWrite into a physical RapidsDeltaWriteExec */
+object RapidsDeltaWriteStrategy extends SparkStrategy {
+  override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+    case p: RapidsDeltaWrite =>
+      Seq(RapidsDeltaWriteExec(planLater(p.child)))
+    case _ => Nil
+  }
+}
+
+/**
+ * This logical node is a placeholder in the plan for where, semantically, a Delta Lake
+ * write occurs. Normally Delta Lake writes are *not* visible in the SQL plan, but we
+ * use this as a workaround for undesired row transitions in the plan when adaptive
+ * execution is enabled. Without it, AdaptiveSparkPlanExec can become the root node
+ * of the physical plan to write into Delta. Without a parent node above
+ * AdaptiveSparkPlanExec, the GPU transition planning does not know whether to make
+ * the result of the adaptive plan GPU columnar or CPU row, so it ends up assuming CPU
+ * row and injecting a columnar-to-row transition at the end of the plan. The Delta
+ * write code cannot directly replace the AdaptiveSparkPlanExec to force it to be
+ * columnar, because it's tied to the QueryExecution which can only be created with
+ * logical plans. Thus our workaround is to inject a node in the logical plan for
+ * the write and ensure the physical node appears to Spark as a write so the
```

Comment:
Again it might be nice to explain how a "write" stops this from happening. But I think if we only want to explain it in one place, instead of three, then here is the better place to do it.